scripts.py

```diff
@@ -132,95 +132,81 @@ def run(self):
-                    metadata_file = os.path.join(root, file)
-                    break
-            if metadata_file:
-                break
-
-        if not metadata_file:
-            raise Exception("No metadata file found")
-
-        # Extract metadata for each book
+    def run(self, data_source_name, metadata_records, epub_directory, cover_directory):
         replacement_policy = ReplacementPolicy(rights=True, links=True, formats=True)
-        with open(metadata_file) as f:
-            metadata_records = MARCExtractor().parse(f, data_source_name)
-            for metadata in metadata_records:
-                primary_identifier = metadata.primary_identifier
-                isbn = primary_identifier.identifier
-
-                uploader = S3Uploader()
-                paths = dict()
-
-                url = uploader.book_url(primary_identifier, "epub")
-                epub_path = os.path.join(directory, isbn + ".epub")
-                paths[url] = epub_path
-
-                epub_link = LinkData(
-                    rel=Hyperlink.OPEN_ACCESS_DOWNLOAD,
-                    href=url,
-                    media_type=Representation.EPUB_MEDIA_TYPE,
-                )
-
-                formats = [FormatData(
-                    content_type=Representation.EPUB_MEDIA_TYPE,
-                    drm_scheme=DeliveryMechanism.NO_DRM,
-                    link=epub_link,
-                )]
-                circulation_data = CirculationData(
-                    data_source_name,
-                    primary_identifier,
-                    links=[epub_link],
-                    formats=formats,
-                )
-                metadata.circulation = circulation_data
-
-                cover_file = isbn + ".jpg"
-                cover_url = uploader.cover_image_url(
-                    data_source_name, primary_identifier, cover_file)
-                cover_path = os.path.join(directory, cover_file)
-                paths[cover_url] = cover_path
-
-                metadata.links.append(LinkData(
-                    rel=Hyperlink.IMAGE,
-                    href=cover_url,
-                    media_type=Representation.JPEG_MEDIA_TYPE,
-                ))
-
-                edition, new = metadata.edition(self._db)
-                metadata.apply(edition, replace=replacement_policy)
-                pool = get_one(self._db, LicensePool, identifier=edition.primary_identifier)
-
-                if new:
-                    print "created new edition %s" % edition.title
-                for link in edition.primary_identifier.links:
-                    if link.rel == Hyperlink.IMAGE:
-                        representation = link.resource.representation
-                        representation.mirror_url = link.resource.url
-                        representation.local_content_path = paths[link.resource.url]
-                        uploader.mirror_one(representation)   
```

Comment:
When the local file this was trying to upload didn't exist, it would throw an exception and stop the script. Now it prints a message and continues.